Classes/Task/IndexItemsTask.php

```diff
@@ -0,0 +1,226 @@
+     *
+     * @return string Information to display
+     */
+    public function getAdditionalInformation(): string
+    {
+        return parent::getAdditionalInformation();
+    }
+
+    /**
+     * Gets initial records
+     *
+     * @param string $settings
+     * @return array
+     * @throws \InvalidArgumentException
+     * @throws \UnexpectedValueException
+     */
+    protected function getInitialRecords(string $settings): array
+    {
+        $resultRows = GeneralUtility::makeInstance(ConnectionPool::class)
+            ->getConnectionForTable('tt_content')
+            ->select(
+                ['pi_flexform'],
+                'tt_content',
+                ['uid' => $this->pluginTtContentUid]
+            )->fetch();
+        $flexform = GeneralUtility::xml2array($resultRows['pi_flexform']);
+        return explode(',', $flexform['data']['sDEFAULT']['lDEF'][$settings]['vDEF']);
+    }
+
+    /**
+     * Gets live data of given records
+     *
+     * @param array $records
+     * @return array
+     */
+    protected function getLiveDataForRecords(array $records): array
+    {
+        $recordsToIndex = [];
+
+        foreach ($records as $recordToIndex) {
+            try {
+                $record = call_user_func(
+                    [$this->repository, $this->classMapping[$this->typeToIndex]['methodLiveById']],
+                    $recordToIndex['id']
+                );
+            } catch (ClientException $exception) {
+
+            }
+
+            // TODO: Search can be optimized by imploding for sub arrays in sections like address
+            if ($record['sections']) {
+                $record['processed_sections'] = $this->resolveSectionsText($record['sections']);
+            }
+            if ($record['organisationseinheit']) {
+                $record['processed_organisationseinheit'] = $this->multi_implode($record['organisationseinheit'], ',');
+            }
+            if ($record['preamble']) {
+                $record['preamble'] = strip_tags($record['preamble']);
+            }
+
+            $recordsToIndex[] = $record;
+            if ($recordToIndex['_children']) {
+                $recordsToIndex = array_merge($recordsToIndex, $this->getLiveDataForRecords($recordToIndex['_children']));
+            }
+        }
+
+        return $recordsToIndex;
+    }
+
+    /**
+     * Resolve sections text
+     *
+     * @param array $sections
+     * @return string
+     */
+    protected function resolveSectionsText(array $sections): string
+    {
+        $result = '';
+        foreach ($sections as $section) {
+            if ($section['text']) {
+                $result .= $section['text'] . ',';
+            }
+        }
+
+        return strip_tags(rtrim($result, ','));
+    }
+
+    /**
+     * Multi implode
+     *
+     * @param $array
+     * @param $glue
+     * @return bool|string
+     */
+    protected function multi_implode($array, $glue): string
+    {
+        $ret = '';
```

Comment:
Can you please use a more describing variable name like $content or $implodedString, or what ever...